packages/nx/src/native/native_hasher.rs

```diff
@@ -97,6 +99,75 @@ fn hash_files(workspace_root: String) -> HashMap<String, String> {
     receiver_thread.join().unwrap()
 }
 
+#[napi]
+fn hash_files_matching_globs(directory: String, glob_patterns: Vec<String>) -> anyhow::Result<Option<String>> {
+    let mut globset_builder = GlobSetBuilder::new();
+
+    for pattern in glob_patterns {
+        globset_builder.add(
+            Glob::new(&pattern)
+                    .map_err(|_| anyhow!("Invalid Glob {pattern}"))?);
+    }
+    let globset = globset_builder.build().map_err(|_| anyhow!("Error building globset builder"))?;
+
+    let cpus = available_parallelism().map_or(2, |n| n.get()) - 1;
+
+    let mut walker = WalkBuilder::new(&directory);
+    walker.hidden(false);
+
+    let (sender, receiver) = unbounded::<(String, Vec<u8>)>();
+
+    let receiver_thread = thread::spawn(move || {
+        let mut collection: Vec<String> = Vec::new();
```

Comment:
This is not correct, the order of these hashes must be deterministic. This will need to return the `Vec<FileData>`